go/vt/sqlparser/ast.go

```diff
@@ -251,21 +251,35 @@ func (node *Union) WalkSubtree(visit Visit) error {
 	)
 }
 
-// Insert represents an INSERT statement.
+// Insert represents an INSERT or REPLACE statement.
+// Per the MySQL docs, http://dev.mysql.com/doc/refman/5.7/en/replace.html
+// Replace is the counterpart to `INSERT IGNORE`, and works exactly like a
+// normal INSERT except if the row exists. In that case it first deletes
+// the row and re-inserts with new values. For that reason we keep it as an
+// Insert struct and special case it using the Replace bool.
+// Replaces are currently disallowed in sharded schemas because
+// of the implications the deletion part may have on vindexes.
 type Insert struct {
 	Comments Comments
 	Ignore   string
 	Table    *TableName
 	Columns  Columns
 	Rows     InsertRows
 	OnDup    OnDup
+	Replace  bool
```

Comment:
For uniformity, I'd recommend using `Operation` or `Command` as a field, that's initialized to `"insert"` or `"replace"`. This way, you won't need a conditional at the time of printing. You'll see similar examples for this in operators, distinct, etc.